python/lsst/ctrl/bps/htcondor/prepare_utils.py

```diff
@@ -101,11 +101,12 @@ def _create_job(subdir_template, cached_values, generic_workflow, gwjob, out_pre
         "getenv": "True",
         # Exceeding memory sometimes triggering SIGBUS or SIGSEGV error. Tell
         # htcondor to put on hold any jobs which exited by a signal.
-        "on_exit_hold": "ExitBySignal == true",
-        "on_exit_hold_reason": (
-            'strcat("Job raised a signal ", string(ExitSignal), ". ", '
-            '"Handling signal as if job has gone over memory limit.")'
-        ),
+        "on_exit_hold": "ExitBySignal == true || ExitCode > 128",
+        "on_exit_hold_reason": "ExitBySignal == true ? "
+        'strcat("Job raised a signal ", string(ExitSignal), '
+        '". Handling signal as if job has gone over memory limit.") : '
+        'strcat("Job exit code (", string(ExitCode), ") > 128. '
+        'Handling signal as if job has gone over memory limit.")',
```

Comment:
Can we update the comment so it also explains why `ExitCode > 128` is there? Something like

> Tell htcondor to put on hold jobs that may have failed because they exceeded their memory allocation. When it happens regular jobs usually trigger SIGBUS or SIGSEGV error or finish with exit code above 128.